### DISCUSSION TITLE
[Secrets Vars] Multiple line variables management

### DISCUSSION THREAD

**Discussion by ghost** [mia-platform/community#374] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v12.0.1

### Description

Hi guys!
I need to setup a secret variable that has multiple lines, in particular, this variable contains a Public Key, so a new line is required after a set number of characters. The Console let me add the value with the appropriate new lines, but when I deploy the service that uses this variable, I can see from the deployment file that it is not added as expected.

What is the best way to handle this kind of values? It is a issue related on how the Console and mlp handles them, or it is implied that I have to manage this values in the service?

In the Actual vs Expected outcome an example of deployment file that uses a new line variable after it is applied to the cluster by mlp.
Thank you very much!

### Actual Outcome

        - name: PUBLIC_KEY
          value: '-----BEGIN PUBLIC KEY-----\nsome_base64\nsome_base64\nsome_base64\nsome_base64\nsome_base64\nsome_base64\n-----END
            PUBLIC KEY-----'

### Expected Outcome

        - name: PUBLIC_KEY
          value: |-
            -----BEGIN PUBLIC KEY-----
            some_base64
            some_base64
            some_base64
            some_base64
            some_base64
            some_base64
            some_base64
            -----END PUBLIC KEY-----

or, alternatevely:

        - name: PUBLIC_KEY
          value: "-----BEGIN PUBLIC KEY-----\nsome_base64\nsome_base64\nsome_base64\nsome_base64\nsome_base64\nsome_base64\n-----END
            PUBLIC KEY-----"

So with the double quotes, since the manifest will automatically update with the first version in the Expected outcome.

**Comment by flower-of-the-bridges** [comment, marked as answer]
Hi @ivanpasquariello 

To handle certificates in my project I've chosen to declare a secret in `mlp.yaml` where a `.pem` file is created

```yaml
secrets:
  - name: "my-certificate"
    when: "always"
    data:
      - from: "literal"
        key: "my-certificate-.pem"
        value: "{{PUBLIC_KEY}}"
``` 

Then I can reference It in my microservices. However, this means that the microservices involved have to handle the `.pem` file.

I'm interested too in finding some alternatives, so I'm following this thread too 👓
